src/migrations/20240610110823-remove-unwanted-objectives.js

```diff
@@ -3,68 +3,120 @@ const {
+            AS
+      WITH updateresources AS (
+      DELETE FROM "ActivityReportObjectiveResources" WHERE "activityReportObjectiveId" IN (
+        SELECT id FROM "ActivityReportObjectives" WHERE "objectiveId" IN (
+          SELECT id FROM ObjectivesToRemove
+        )
+      )
+      RETURNING
+      id
+      ) SELECT * FROM updateresources;
+
+      -- Delete from ARO Files.
+      DROP TABLE IF EXISTS DeleteFiles;
+      CREATE TEMP TABLE DeleteFiles
+            AS
+      WITH updatedfiles AS (
+      DELETE FROM "ActivityReportObjectiveFiles" WHERE "activityReportObjectiveId" IN (
+        SELECT id FROM "ActivityReportObjectives" WHERE "objectiveId" IN (
+          SELECT id FROM ObjectivesToRemove
+        )
+      )
+      RETURNING
+      id
+      ) SELECT * FROM updatedfiles;
+
+      -- Delete from AR Courses.
+      DROP TABLE IF EXISTS DeleteCourses;
+      CREATE TEMP TABLE DeleteCourses
+            AS
+      WITH updatedcourses AS (
+      DELETE FROM "ActivityReportObjectiveCourses" WHERE "activityReportObjectiveId" IN (
+        SELECT id FROM "ActivityReportObjectives" WHERE "objectiveId" IN (
+          SELECT id FROM ObjectivesToRemove
+        )
+      )
+      RETURNING
+      id
+      ) SELECT * FROM updatedcourses;
 
       -- Delete ARO's.
+      DROP TABLE IF EXISTS DeleteAROs;
+      CREATE TEMP TABLE DeleteAROs
+            AS
+      WITH updatearos AS (
       DELETE FROM "ActivityReportObjectives" WHERE "objectiveId" IN (
-        SELECT id FROM "Objectives" WHERE id IN (
-            179385,
-            179427,
-            179494,
-            179500,
-            179550,
-            179606,
-            179635,
-            179556,
-            180440,
-            177557) AND "status" = 'Not Started'
-      );
+        SELECT id FROM ObjectivesToRemove
+      )
+      RETURNING
+      id
+      ) SELECT * FROM updatearos;
+
+      -- Delete objectives.
+      DROP TABLE IF EXISTS DeleteObjectives;
+      CREATE TEMP TABLE DeleteObjectives
+            AS
+      WITH updateobjectives AS (
+      DELETE FROM "Objectives" WHERE id IN (
+         SELECT id FROM ObjectivesToRemove
+      )
+      RETURNING
+      id
+      ) SELECT * FROM updateobjectives;
 
-        -- Delete objectives.
-        DELETE FROM "Objectives" WHERE id IN (
-          179385,
-          179427,
-          179494,
-          179500,
-          179550,
-          179606,
-          179635,
-          179556,
-          180440,
-          177557) AND "status" = 'Not Started';
+      -- Get Delete counts using union.
+      SELECT COUNT(*), 'ARO Topics' FROM DeleteTopics
+      UNION ALL
+      SELECT COUNT(*), 'ARO Resources' FROM DeleteResources
+      UNION ALL
+      SELECT COUNT(*), 'ARO Files' FROM DeleteFiles
+      UNION ALL
+      SELECT COUNT(*), 'ARO Courses' FROM DeleteCourses
+      UNION ALL
+      SELECT COUNT(*), 'AROs' FROM DeleteAROs
+      UNION ALL
+      SELECT COUNT(*), 'Objectives' FROM DeleteObjectives;
```

Comment:
please add a drop for all the temp tables created, normally this would not be needed as they are automatically dropped at the end of the transaction. But it has been a pain as of late that subsequent migrations are still part of the same transaction and thus these temp tables can effect how later migration files run.